frontera/contrib/backends/cassandra/__init__.py

```diff
@@ -0,0 +1,201 @@
+        return Queue(self.session_cls, self.models['QueueModel'], settings.get('SPIDER_FEED_PARTITIONS'),
+                     ordering='created_desc')
+
+BASE = CommonBackend
+LIFO = LIFOBackend
+FIFO = FIFOBackend
+
+
+class Distributed(DistributedBackend):
+    def __init__(self, manager):
+        self.manager = manager
+        settings = manager.settings
+        cluster_ips = settings.get('CASSANDRABACKEND_CLUSTER_IPS')      # Format: ['192.168.0.1', '192.168.0.2']
+        cluster_port = settings.get('CASSANDRABACKEND_CLUSTER_PORT')
+        keyspace = settings.get('CASSANDRABACKEND_KEYSPACE')
+        keyspace_create = settings.get('CASSANDRABACKEND_CREATE_KEYSPACE_IF_NOT_EXISTS')                # Default: true
+        models = settings.get('CASSANDRABACKEND_MODELS')
+
+        self.cluster = Cluster(cluster_ips, cluster_port)
+        self.models = dict([(name, load_object(klass)) for name, klass in models.items()])
+
+        self.session_cls = self.cluster.connect()
+        self.session_cls.row_factory = dict_factory
+
+        if keyspace_create:
+            query = """CREATE KEYSPACE IF NOT EXISTS \"%s\"
+                        WITH replication = {'class': 'SimpleStrategy', 'replication_factor' : 3}""" % (keyspace, )
+            self.session_cls.execute(query)
+        self.session_cls.set_keyspace(keyspace)
+        connection.set_session(self.session_cls)
+
+        self._metadata = None
+        self._queue = None
+        self._states = None
+
+    @classmethod
+    def strategy_worker(cls, manager):
+        b = cls(manager)
+        settings = manager.settings
+        drop_all_tables = settings.get('CASSANDRABACKEND_DROP_ALL_TABLES')
+        model = b.models['StateModel']
+
+        if drop_all_tables:
+            drop_table(model)
+
+        sync_table(model)
+
+        b._states = States(b.session_cls, model,
+                           settings.get('STATE_CACHE_SIZE_LIMIT'))
+        return b
+
+    @classmethod
+    def db_worker(cls, manager):
+        b = cls(manager)
+        settings = manager.settings
+        drop = settings.get('CASSANDRABACKEND_DROP_ALL_TABLES')
+
+        metadata_m = b.models['MetadataModel']
+        queue_m = b.models['QueueModel']
+        stats_m = b.models['CrawlStatsModel']
+        if drop:
+            drop_table(metadata_m)
+            drop_table(queue_m)
+            drop_table(stats_m)
+
+        sync_table(metadata_m)
+        sync_table(queue_m)
+        sync_table(stats_m)
+
+        b._metadata = Metadata(b.session_cls, metadata_m,
+                               settings.get('CASSANDRABACKEND_CACHE_SIZE'))
+        b._queue = Queue(b.session_cls, queue_m, settings.get('SPIDER_FEED_PARTITIONS'))
+        return b
+
+    @property
+    def queue(self):
+        return self._queue
+
+    @property
+    def metadata(self):
+        return self._metadata
+
+    @property
+    def states(self):
+        return self._states
+
+    def frontier_start(self):
```

Comment:
are you sure all these methods need to be copy/pasted here? IMO, it would be great to extract them to parent class.